statsmodels/multivariate/multivariate_ols.py

```diff
@@ -0,0 +1,517 @@
+                terms = self.design_info.term_name_slices
+                hypotheses = []
+                for key in terms:
+                    L_contrast = np.eye(k_xvar)[terms[key], :]
+                    hypotheses.append([key, L_contrast, None])
+            else:
+                hypotheses = []
+                for i in range(k_xvar):
+                    name = 'x%d' % (i)
+                    L = np.zeros([1, k_xvar])
+                    L[i] = 1
+                    hypotheses.append([name, L, None])
+
+        results = _multivariate_ols_test(hypotheses, self._fittedmod,
+                                          self.exog_names, self.endog_names)
+
+        return MultivariateTestResults(results)
+    mv_test.__doc__=(
+        """
+        Testing the linear hypotheses
+            L * params * M = C
+        where `params` is the regression coefficient matrix for the
+        linear model y = x * params, `L` is the contrast matrix, `M` is the
+         dependent variable transform matrix and C is the constant matrix.
+
+        Parameters
+        ----------
+        """ + _hypotheses_doc +
+        """
+        Returns
+        -------
+        results: _MultivariateOLSResults
+
+        """)
+    def summary(self):
+        raise NotImplementedError
+
+
+class MultivariateTestResults(object):
+    def __init__(self, mv_test_df):
+        self.results = mv_test_df
+
+    def __str__(self):
+        return self.summary().__str__()
+
+    def __getitem__(self, item):
+        return self.results[item]
+
+    @property
+    def summary_frame(self):
+        """
+        Return results as a multiindex dataframe
+        """
+        df = []
+        for key in self.results:
+            tmp = self.results[key]['stat'].copy()
+            tmp.loc[:, 'Effect'] = key
+            df.append(tmp.reset_index())
+        df = pd.concat(df, axis=0)
+        df = df.set_index(['Effect', 'index'])
+        df.index.set_names(['Effect', 'Statistic'], inplace=True)
+        return df
+
+    def summary(self, contrast_L=False, transform_M=False):
+        """
+
+        Parameters
+        ----------
+        contrast_L : True or False
+            Whether to show contrast_L matrix
+        transform_M : True or False
+            Whether to show transform_M matrix
+        """
+        summ = summary2.Summary()
+        summ.add_title('Multivariate linear model')
+        for key in self.results:
+            summ.add_dict({'':''})
+            df = self.results[key]['stat'].copy()
+            df = df.reset_index()
+            c = df.columns.values
+            c[0] = key
+            df.columns = c
+            df.index = ['', '', '', '']
+            summ.add_df(df)
+            if contrast_L:
+                summ.add_dict({key:' contrast L='})
+                df = pd.DataFrame(self.results[key]['contrast_L'],
+                                  columns=self.exog_names)
+                summ.add_df(df)
```

Comment:
Does this really work?   contrast and transform have different number of columns. Does this print a new table?